app.js

```diff
@@ -36,41 +38,109 @@ store.on("error", function (error) {
 const inProduction = app.get("env") === "production";
 const sessionParms = {    
     secret: process.env.SESSION_SECRET,// The secret used to sign the session ID cookie    
-    resave: true,// Forces the session to be saved back to the session store, even if the session was never modified during the request
-    
-    saveUninitialized: true,// Forces a session that is "uninitialized" to be saved to the store
-    
-    store: store,// The store instance where sessions will be stored
-    
-    cookie: { secure: false, sameSite: "strict", secure: inProduction  },// Options for the session ID cookie
+    resave: true,// Forces the session to be saved back to the session store, even if the session was never modified during the request    
+    saveUninitialized: true,// Forces a session that is "uninitialized" to be saved to the store    
+    store: store,// The store instance where sessions will be stored    
+    cookie: { sameSite: "strict", secure: inProduction  },// Options for the session ID cookie
 };
   
+const csrf_options = {
+    protected_operations: ["PATCH"],
+    protected_content_types: ["application/json"],
+    development_mode:  !inProduction,
+    cookie: true,
+};
+
 // If the application is in production
 if (inProduction) {    
     app.set("trust proxy", 1); // trust first proxy       
 }
   
 // Use the session middleware with the defined parameters
 app.use(session(sessionParms));
+app.set("view engine", "ejs");// Set the view engine to ejs for rendering views
+app.use(bodyParser.urlencoded({ extended: true })); // Use body-parser middleware to parse incoming request bodies
+app.use(flash()); // Add the connect-flash middleware
+app.use(cookieParser(process.env.SESSION_SECRET));
+app.use(express.urlencoded({ extended: false }));
+// app.use(csurf());
+app.use(methodOverride('_method'));
+
+// Security middlewares
+app.use(helmet());
+app.use(xssClean());
+const limiter = rateLimit({
+  windowMs: 15 * 60 * 1000, // 15 minutes
+  max: 100 // limit each IP to 100 requests per windowMs
+});
+app.use(limiter);
 
 // Add the passport middleware
-
 passportInit();
 app.use(passport.initialize());
 app.use(passport.session());
 
-// Add the connect-flash middleware
-app.use(require("connect-flash")());
 
-app.use(require("./middleware/storeLocals"));
-app.get("/", (req, res) => {
-    res.render("index");
+
+app.use(csrf());
```

Comment:
Requires change:
Since this csrf protection is being used for EVERY route, you don't need individual `csrfProtection` middleware for your routes. You'll only need one: global middleware or per route middleware. In this case, since you don't need it globally for every route, I'd recommend extracting the below into a middleware `csrfProtection.js` file and use it where needed:
```js
const csrf = require('csurf');

const csurfOptions = {
    protected_operations: ["PATCH"],
    protected_content_types: ["application/json"],
    development_mode:  !inProduction,
    cookie: true,
};

const csrfProtection = csurf(csurfOptions);

module.exports = csrfProtection;
```

Then you use require this middleware file and use it in your `jobs` and `sessionController` routes:
```js
const express = require("express");
const router = express.Router();
const csrfProtection = require("../middleware/csrfProtection");

router.route("/jobs/delete/:id", csrfProtection, deleteJob);
// other "CUD" routes that change data via a form submission
```